File: CH-Academy/composeApp/Utils/Firebase/AuthRepository.swift
import Foundation
import Firebase

@objc class IOSAuthRepository: NSObject {

    @objc func requestOTP(phoneNumber: String, onSuccess: @escaping (String) -> Void, onError: @escaping (NSError) -> Void) {
        PhoneAuthProvider.provider().verifyPhoneNumber(phoneNumber, uiDelegate: nil) { verificationID, error in
            if let error = error as NSError? {
                onError(error)
            } else {
                onSuccess(verificationID ?? "")
            }
        }
    }

    @objc func verifyOTP(verificationId: String, code: String, onSuccess: @escaping () -> Void, onError: @escaping (NSError) -> Void) {
        let credential = PhoneAuthProvider.provider().credential(withVerificationID: verificationId, verificationCode: code)
        Auth.auth().signIn(with: credential) { _, error in
            if let error = error as NSError? {
                onError(error)
            } else {
                onSuccess()
            }
        }
    }
}

File: CH-Academy/composeApp/src/commonMain/kotlin/com/chacademy/android/Login/AuthRepository.kt
package com.chacademy.android.Login

interface AuthRepository {
    fun requestOTP(phoneNumber: String, onSuccess: (verificationId: String) -> Unit, onError: (exception: Exception) -> Unit)
    fun verifyOTP(verificationId: String, code: String, onSuccess: () -> Unit, onError: (exception: Exception) -> Unit)
    fun resendOTP(phoneNumber: String, onSuccess: (verificationId: String) -> Unit, onError: (exception: Exception) -> Unit)
}